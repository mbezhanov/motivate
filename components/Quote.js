import React from 'react';
import { View } from 'react-native';
import StyledText from './StyledText';
import Sizes from '../constants/Sizes';
import PropTypes from 'prop-types';

const quote = ({ quote }) => {
  if (!quote) {
    return <View/>;
  }

  return (
    <View>
      <StyledText size={Sizes.XL}>{quote.content}</StyledText>
      <StyledText size={Sizes.L}>{quote.author}</StyledText>
      <StyledText size={Sizes.M}>{quote.book}</StyledText>
    </View>
  );
};

quote.propTypes = {
  quote: PropTypes.shape({
    content: PropTypes.string,
    author: PropTypes.string,
    book: PropTypes.string,
  }),
};

export default quote;
