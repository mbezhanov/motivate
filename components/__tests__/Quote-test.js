import React from 'react';
import renderer from 'react-test-renderer';
import Quote from '../Quote';

jest.mock('../StyledText', () => 'StyledText');

describe('Quote', () => {
  it('renders correctly', () => {
    const quote = {
      content: 'foo',
      author: 'bar',
      book: 'baz',
    };
    const tree = renderer.create(<Quote quote={quote} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders an empty View if no "quote" property is specified', () => {
    const tree = renderer.create(<Quote />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
