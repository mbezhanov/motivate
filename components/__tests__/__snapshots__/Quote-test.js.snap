// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote renders an empty View if no "quote" property is specified 1`] = `<View />`;

exports[`Quote renders correctly 1`] = `
<View>
  <StyledText
    size={28}
  >
    foo
  </StyledText>
  <StyledText
    size={21}
  >
    bar
  </StyledText>
  <StyledText
    size={16}
  >
    baz
  </StyledText>
</View>
`;
